db/migrate/20200210150200_associate_move_with_profile_part2.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+class AssociateMoveWithProfilePart2 < ActiveRecord::Migration[5.2]
+  def up
+    Move.find_each.reject { |m| m.profile_id.present? }.each do |move|
```

Comment:
This will instantiate every move record in the the system _and then_ check if it has a `profile_id`. Instead use the database to filter the records first, e.g.
```
Move.where(profile_id: nil).each do |move|
```

Reply:
This is idiomatic for migrations. Your proposal will instantiate every move record (on first run). ActiveRecord::Base#find_each maps to ActiveRecord::Base#find_in_batches (which discards the records after they have been iterated)

Reply:
hmmm, as a compromise how about:

    Move.where(profile_id: nil).find_each do |move|

That will use the efficiency of the database and won't keep all records in memory.

Reply:
I'm not sure that works? find_each goes through the whole table in ID order

Reply:
hi @starswan yes I think it works:

 ```
pry(main)> Move.where(profile_id: nil).find_each { |m| puts m }
  Move Load (3.7ms)  SELECT  "moves".* FROM "moves" WHERE "moves"."profile_id" IS NULL ORDER BY "moves"."id" ASC LIMIT $1  [["LIMIT", 1000]]
#<Move:0x00007faaf50116f0>
#<Move:0x00007faaf5010390>
#<Move:0x00007faaf5010200>
#<Move:0x00007faaf50100c0>
#<Move:0x00007faaf501bf38>
...
```

Reply:
I mean I don't think it is memory efficient